test/UnitTests/Microsoft.OData.Core.Tests/Json/ODataJsonEntryAndFeedDeserializerUndeclaredTests.cs

```diff
@@ -1185,28 +1212,51 @@ public void ReadOpenEntryUndeclaredEmptyCollectionPropertiesWithoutODataTypeAsOD
                             complex1 = (reader.Item as ODataResource);
                         }
                     }
+                    else if (reader.State == ODataReaderState.ResourceSetStart)
+                    {
+                        Assert.NotNull(nestedResourceInfo);
+                        Assert.Equal("UndeclaredCollection1", nestedResourceInfo.Name);
+                        Assert.Null(undeclaredCollection1Values); // guard
+                        undeclaredCollection1Values = new List<ODataPrimitiveValue>();
+                    }
+                    else if (reader.State == ODataReaderState.Primitive)
+                    {
+                        Assert.NotNull(undeclaredCollection1Values);// guard
+                        undeclaredCollection1Values.Add(reader.Item as ODataPrimitiveValue);
+                    }
+                    else if (reader.State == ODataReaderState.NestedResourceInfoStart)
+                    {
+                        nestedResourceInfo = (ODataNestedResourceInfo)reader.Item;
+                    }
+                    else if (reader.State == ODataReaderState.NestedResourceInfoEnd)
+                    {
+                        nestedResourceInfo = null;
+                    }
                 },
                 settings: new ODataMessageReaderSettings
                 {
                     ShouldIncludeAnnotation = (annotationName) => true,
                     Validations = ~ValidationKinds.ThrowOnUndeclaredPropertyForNonOpenType,
-                    EnableUntypedCollections = true,
+                    EnableUntypedCollections = true, // Sam: This configuration looks unnecessary? We should consider to remove this configuration when removing/obsolote 'ODataUntypedValue'

```

Comment:
Did you create a pull request to mark `ODataUntypedValue` as obsolete? Or should we deprecate it in ODL 9 and remove it in ODL 10?